app/models/apple/api_error.rb

```diff
@@ -22,5 +22,28 @@ def format_message(message, response)
 
       resp
     end
+
+    def self.for_response(message, response)
+      begin
+        body = JSON.parse(response.body) if response.try(:body).present?
+
+        # Check API key permission error pattern
+        if matches_permission_error_pattern?(body)
+          return ApiPermissionError.new("Apple API permission error - delegated delivery key lacks required permissions", response)
+        end
+      rescue JSON::ParserError
+        # Fall through to default
+      end
+
+      new(message, response)
+    end
+
+    def self.matches_permission_error_pattern?(body)
+      return false unless body.is_a?(Hash)
+
+      first_error = body&.dig("errors", 0)
+      first_error&.dig("code") == "FORBIDDEN_ERROR" &&
```

Comment:
Have you seen forbidden statuses for other problems?  Just wondering if matching on the error details could break if they ever reworded it slightly.